src/qcodes/parameters/delegate_parameter.py

```diff
@@ -223,6 +227,23 @@ def unit(self) -> str:
     def unit(self, unit: str | None) -> None:
         self._unit_override = unit
 
+    @property
+    def vals(self) -> Validator | None:
+        """
+        The validator of the parameter. Read from source if not explicitly overwritten.
+        Set to None to disable overwrite.
+        """
+        if self._vals_override is not None:
+            return self._vals_override
+        elif self.source is not None:
+            return self.source.vals
+        else:
+            return None
```

Comment:
ParameterBase supports multiple validators via the add_validator etc logic. We need to consider if we want to support that here too.